File: AgoraChatSDK/CWrapper/Wrapper/Wrapper_Android/JavaWrapperDemo/wrapper/src/main/java/com/hyphenate/wrapper/EMWrapper.java
package com.hyphenate.wrapper;

import com.hyphenate.helper.EMUnityHelper;
import com.hyphenate.wrapper.util.EMSDKMethod;
import com.hyphenate.wrapper.callback.EMWrapperCallback;

import org.json.JSONException;
import org.json.JSONObject;

public class EMWrapper {
    public EMClientWrapper clientWrapper;
    public EMWrapper() {
        clientWrapper = new EMClientWrapper();
        new EMUnityHelper();
    }

    public String callSDKApi(String manager, String method, String jsonString, EMWrapperCallback callback) {
        String str = null;
        try {
            JSONObject jsonObject = null;
            if (jsonString.length() > 0) {
                jsonObject = new JSONObject(jsonString);
            }
            switch (manager) {
                case EMSDKMethod.client:
                    str = clientWrapper.onMethodCall(method, jsonObject, callback);
                    break;
                case EMSDKMethod.chatManager:
                    str = clientWrapper.chatManagerWrapper.onMethodCall(method, jsonObject, callback);
                    break;
                case EMSDKMethod.contactManager:
                    str = clientWrapper.contactManagerWrapper.onMethodCall(method, jsonObject, callback);
                    break;
                case EMSDKMethod.roomManager:
                    str = clientWrapper.roomManagerWrapper.onMethodCall(method, jsonObject, callback);
                    break;
                case EMSDKMethod.groupManager:
                    str = clientWrapper.groupManagerWrapper.onMethodCall(method, jsonObject, callback);
                    break;
                case EMSDKMethod.userInfoManager:
                    str = clientWrapper.userInfoManagerWrapper.onMethodCall(method, jsonObject, callback);
                    break;
                case EMSDKMethod.presenceManager:
                    str = clientWrapper.presenceManagerWrapper.onMethodCall(method, jsonObject, callback);
                    break;
                case EMSDKMethod.threadManager:
                    str = clientWrapper.chatThreadManagerWrapper.onMethodCall(method, jsonObject, callback);
                    break;
                case EMSDKMethod.pushManager:
                    str = clientWrapper.pushManagerWrapper.onMethodCall(method, jsonObject, callback);
                    break;
            }
        } catch (JSONException e) {
            try {
                JSONObject jsonObject = new JSONObject();
                jsonObject.put("error", e.getLocalizedMessage());
                callback.onError(jsonObject);
            }catch (JSONException ignore){}
        }
        return str;
    }
}

File: AgoraChatSDK/CWrapper/Wrapper/Wrapper_iOS/wrapper/wrapper/EMWrapper.m
//
//  EMWrapper.m
//  wrapper
//
//  Created by 杜洁鹏 on 2022/11/11.
//

#import "EMWrapper.h"
#import "EMClientWrapper.h"
#import "EMSDKMethod.h"

@interface EMWrapper ()
@property (nonnull, strong) EMClientWrapper *clientWrapper;
@end

@implementation EMWrapper

- (instancetype)init {
    if (self = [super init]) {
        _clientWrapper = [EMClientWrapper shared];
    }
    return self;
}

- (NSString *)callSDKApi:(NSString *)manager method:(NSString *)method params:(NSDictionary *)params callback:(EMWrapperCallback *)callback {
    NSString *str = nil;
    if ([manager isEqualToString:client]) {
        str = [_clientWrapper onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:chatManager]) {
        str = [_clientWrapper.chatManager onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:contactManager]) {
        str = [_clientWrapper.contactManagerWrapper onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:roomManager]) {
        str = [_clientWrapper.roomManagerWrapper onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:groupManager]) {
        str = [_clientWrapper.groupManagerWrapper onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:userInfoManager]) {
        str = [_clientWrapper.userInfoManagerWrapper onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:presenceManager]) {
        str = [_clientWrapper.presenceManagerWrapper onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:threadManager]) {
        str = [_clientWrapper.chatThreadManagerWrapper onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:pushManager]) {
        str = [_clientWrapper.pushManagerWrapper onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:messageManager]) {
        str = [_clientWrapper.messageManager onMethodCall:method params:params callback:callback];
    }else if ([manager isEqualToString:conversationManager]) {
        str = [_clientWrapper.conversationWrapper onMethodCall:method params:params callback:callback];
    }
    return str;
}
@end
